### PR TITLE
[NO SQUASH] Fixing unit tests and adding project-specific SonarLint.xml in ITs

### PR DESCRIPTION
**Pull request by valhristov**

### CODE REVIEW

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj [hunk 1]
```diff
@@ -199,10 +199,6 @@
     <Compile Include="..\..\src\AssemblyInfo.Shared.cs">
       <Link>Properties\AssemblyInfo.Shared.cs</Link>
     </Compile>
-    <None Include="TestCases\TestMethodShouldNotBeIgnored.NUnit.V2.cs">
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj]: Crap...

File: sonaranalyzer-dotnet/its/SonarAnalyzer.Testing.ImportBefore.targets [hunk 2]
```diff
@@ -3,14 +3,21 @@
           AfterTargets="ResolveCodeAnalysisRuleSet"
           BeforeTargets="CoreCompile">
 
+    <!-- Copy project-specific SonarLint.xml to enable project-specific settings, such as file headers -->
+    <PropertyGroup>
+      <ProjectSpecificSonarLintXml Condition="Exists('$(MSBuildStartupDirectory)\config\$(PROJECT)\SonarLint.xml')">
+        $(MSBuildStartupDirectory)\config\$(PROJECT)\SonarLint.xml
+      </ProjectSpecificSonarLintXml>
+      <ProjectSpecificSonarLintXml Condition="!Exists('$(MSBuildStartupDirectory)\config\$(PROJECT)\SonarLint.xml')">
```
Comment (duncanp-sonar) [inline sonaranalyzer-dotnet/its/SonarAnalyzer.Testing.ImportBefore.targets]: Minor - not worth changing, but this condition could just be   ...='' . It makes it a bit more obvious that it is setting a default/fallback value, and it is also easier to maintain in the future if the condition ever changes.
The other pattern setting a default is to unconditionally set the default value first, then follow it with a property definition with a condition that will overwrite it.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (duncanp-sonar) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj]: Sounds like a good reason for having all non-trivial changes reviewed.

[the remark this answers is not in the document]
Comment (Evangelink) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj]: Sounds like a good reason to have a tool to ensure we don't change this OR to finally migrate the unit test project to the new format.

**Pull request merged**